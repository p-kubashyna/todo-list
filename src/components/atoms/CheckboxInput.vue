<script setup>
const props = defineProps({
  checked: {
    type: Boolean,
    required: true,
    default: false
  }
})
</script>

<template>
  <input type="checkbox" class="checkbox-round" :checked="props.checked"/>
</template>

<style>
.checkbox-round {
  width: 1em;
  height: 1em;
  background-color: white;
  border-radius: 50%;
  vertical-align: middle;
  border: 1px solid #5b5b5c;
  appearance: none;
  -webkit-appearance: none;
  outline: none;
  cursor: pointer;
}

.checkbox-round:checked {
  background-color: #5b5b5c;
}

.checkbox-round:hover {
  border-width: 2px;
}
</style>
