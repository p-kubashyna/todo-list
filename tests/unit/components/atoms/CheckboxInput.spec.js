import { describe, expect, it } from "@jest/globals";
import { shallowMount } from "@vue/test-utils";
import CheckboxInput from "@/components/atoms/CheckboxInput.vue";

describe("Given the CheckboxInput component", () => {
  it.each([[{ checked: true }], [{ checked: false }]])(
    "should render",
    (propsData) => {
      const wrapper = shallowMount(CheckboxInput, { props: propsData });

      expect(wrapper.element).toMatchSnapshot();
    }
  );
});
