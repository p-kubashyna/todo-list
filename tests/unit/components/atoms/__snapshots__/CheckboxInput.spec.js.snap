// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the CheckboxInput component should render 1`] = `
<input
  checked=""
  class="checkbox-round"
  type="checkbox"
/>
`;

exports[`Given the CheckboxInput component should render 2`] = `
<input
  class="checkbox-round"
  type="checkbox"
/>
`;
